client/pages/projects/index.jsx

```diff
@@ -14,62 +14,276 @@ import {
   Col,
   Container,
   Input,
-  Row
+  Row,
+  Alert
 } from "reactstrap";
 
 import "../../public/styles/projects.scss";
+import "../../../api/public/schema/projectSchema";
 
-const DESCRIPTION_LENGTH = 200;
+import Fuse from "fuse.js";
+
+const DESCRIPTION_LENGTH = 150;
 
 export default function ProjectsPage() {
+  const [allProjects, setAllProjects] = useState(null);
   const [projects, setProjects] = useState(null);
+  const [visAlert, setAlert] = useState(false);
+  const [selectedUNGoals, setSelectedUNGoals] = useState(null);
+  const [selectedCountry, setSelectedCountry] = useState(null);
+  const [selectedGrpSize, setSelectedGrpSize] = useState(null);
+  const [selectedDifficulty, setSelectedDifficulty] = useState(null);
+
+  const [userInput, setUserInput] = useState("");
+
+  const handleUNGoals = selectedUNGoals => {
+    setSelectedUNGoals(selectedUNGoals);
+  };
+
+  const handleCountry = selectedCountry => {
+    setSelectedCountry(selectedCountry);
+  };
+
+  const handleGrpSize = selectedGrpSize => {
+    setSelectedGrpSize(selectedGrpSize);
+  };
+
+  const handleDifficulty = selectedDifficulty => {
+    setSelectedDifficulty(selectedDifficulty);
+  };
+
+  const handleSearchChange = userInput => {
+    setUserInput(userInput.target.value);
+  };
+
+  const getSearchBarText = () => {
+    const text = document.getElementById("user-input").value;
+
+    return text;
+  };
+
+  const options = {
+    keys: ["name", "description"]
+  };
+
+  const populateAllProjects = async () => {
+    const response = await getModels(null, true);
+    setAllProjects(response.data);
+    setProjects(response.data);
+  };
 
   useEffect(() => {
-    const populateProjects = async () => {
-      const numProjects = 21;
+    populateAllProjects();
+  }, []);
```

Comment:
it's generally better practice to define async methods for use in `useEffect` (such as `populateAllProjects`) inside the `useEffect` function itself, as the initialization behavior has some slight quirks